config/datanode.example.toml

```diff
@@ -29,16 +29,38 @@ connect_timeout = "1s"
 # `TCP_NODELAY` option for accepted connections, true by default.
 tcp_nodelay = true
 
-# WAL options, see `standalone.example.toml`.
+# WAL options.
+# Currently, users are expected to choose the wal through the provider field. 
+# When a wal provider is chose, the user should comment out all other wal config 
+# except those corresponding to the chosen one.
 [wal]
 # WAL data directory
+provider = "raft_engine"
+
+# Raft-engine wal options, see `standalone.example.toml`
 # dir = "/tmp/greptimedb/wal"
 file_size = "256MB"
 purge_threshold = "4GB"
 purge_interval = "10m"
 read_batch_size = 128
 sync_write = false
 
+# Kafka wal options.
+# The broker endpoints of the Kafka cluster. ["127.0.0.1:9090"] by default.
+# broker_endpoints = ["127.0.0.1:9090"]
+# Number of topics shall be created beforehand.
+# num_topics = 64
```

Comment:
Why the datanode needs `num_topics`, `num_partitions` and `topic_name_prefix`? They are used in metasrv when allocating the region WAL, but it looks unnecessary for datanode.

Reply:
Yes. Some configs are not necessary. Will remove unnecessary ones when implementing kafka log store.